packages/smithy-client/src/parse-utils.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * Give an input string, strictly parses a boolean value.
+ *
+ * @param value The boolean string to parse.
+ * @returns true for "true", false for "false", otherwise an error is thrown.
+ */
+export const parseBoolean = (value: string): boolean => {
+  switch (value) {
+    case "true":
+      return true;
+    case "false":
+      return false;
+    default:
+      throw new Error(`Unable to parse boolean value "${value}"`);
```

Comment:
should this be TypeError like the `expect` methods, or no?

Reply:
Technically it's not a type error, it's a parse error. We can always change it if necessary

Reply:
As long as at the end of the day we can generate a meaningful SerializationException for callers (when SerializationException has messages... whoops), that's all I care about. Reading messages from bare `Error` types is dangerous, could leak something unintended.